src/test/java/com/cag/cagbackendapi/service/validation/ValidationServiceTest.kt

```diff
@@ -0,0 +1,90 @@
+package com.cag.cagbackendapi.service.validation
+
+import com.cag.cagbackendapi.constants.DetailedErrorMessages
+import com.cag.cagbackendapi.error.exceptions.BadRequestException
+import com.cag.cagbackendapi.error.exceptions.InternalServerErrorException
+import com.nhaarman.mockito_kotlin.*
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.slf4j.Logger
+
+@ExtendWith(MockitoExtension::class)
+internal class ValidationServiceTest {
+
+    @Mock
+    private lateinit var logger: Logger
+
+    @InjectMocks
+    private lateinit var validationService: ValidationService
+
+    @Test
+    fun validateAuthKey_validAuthKey_succeeds() {
+        // assemble
+        val requestAuthKey = "key"
+
+        // act
+        validationService.validateAuthKey(requestAuthKey)
+
+        // assert
+        verifyZeroInteractions(logger)
+    }
+
+    @Test
+    fun validateAuthKey_blankMessage_400BadRequest() {
+        // assemble
+        val inputMessage = ""
+
+        doNothing().whenever(logger).error(DetailedErrorMessages.MISSING_AUTH_KEY)
+
+        // act
+        val actual = assertThrows<BadRequestException> {
+            validationService.validateAuthKey(inputMessage)
+        }
+
+        // assert
+        assertEquals(actual.message, DetailedErrorMessages.MISSING_AUTH_KEY)
+        verify(logger).error(DetailedErrorMessages.MISSING_AUTH_KEY)
+        verifyZeroInteractions(logger)
+    }
+
+    @Test
+    fun validateAuthKey_nullMessage_400BadRequest() {
+        // assemble
+        val inputMessage = null
+
+        doNothing().whenever(logger).error(DetailedErrorMessages.MISSING_AUTH_KEY)
```

Comment:
I want to make sure I understand this line correctly, is it saying to do nothing whenever this error happens?

Reply:
This is just a test to validate that logger is called with the correct values.